backend/controllers/posts.controller.js

```diff
@@ -1,158 +1,190 @@
-        const imageId = images[0].fileId
-        await imageUpload.deleteFile(imageId)
-    } catch (error) {
-        return res.status(500).send({success: false, message: `Error deleting the image: ${error.message}`})
-    }
-}
+  try {
+    const images = await imageUpload.listFiles({ name: imageName });
+    const imageId = images[0].fileId;
+    await imageUpload.deleteFile(imageId);
+  } catch (error) {
+    return res.status(500).send({
+      success: false,
+      message: `Error deleting the image: ${error.message}`,
+    });
+  }
+};
 
 exports.searchForPosts = async (req, res) => {
-    try {
-        const category = req.body.category;
-        const keyword = req.body.keyword;
-        const contentProviders = req.body.contentProviders;
+  try {
+    const category = req.body.category;
+    const keyword = req.body.keyword;
+    const contentProviders = req.body.contentProviders;
 
-        let conditions = []
-        // if keyword is provided
-        if (keyword != null){
-            conditions.push({title: { $regex: new RegExp("" + keyword.toLowerCase(), "gi") }})
-        }
-        // if category is provided
-        if (category != null){
-            conditions.push({category: category})
-        }
-        // if content provider id is provided
-        if (contentProviders.length > 0){
-            conditions.push({postedBy: {$in: contentProviders}})
-        } 
-        
-        const posts = await Post.find({$and: conditions}).populate('postedBy', 'username') // include the author's username
-        return res.status(200).json(posts);
-    } catch (error) {
-        return res.status(500).send({success: false, message: `Server error: ${error.message}`})
+    let conditions = [];
+    // if keyword is provided
+    if (keyword != null) {
+      conditions.push({
+        title: { $regex: new RegExp("" + keyword.toLowerCase(), "gi") },
+      });
+    }
+    // if category is provided
+    if (category != null) {
+      conditions.push({ category: category });
+    }
+    // if content provider id is provided
+    if (contentProviders.length > 0) {
+      conditions.push({ postedBy: { $in: contentProviders } });
     }
-}
+
+    const posts = await Post.find({ $and: conditions }).populate(
+      "postedBy",
+      "username"
+    ); // include the author's username
+    return res.status(200).json(posts);
+  } catch (error) {
+    return res
+      .status(500)
+      .send({ success: false, message: `Server error: ${error.message}` });
+  }
+};
+
+exports.deleteAllPostsByUserId = async (req, res) => {
+  try {
+    const userId = res.locals.userId;
+    const user = await User.findOne({ _id: userId });
+    const deletedPosts = await Post.deleteMany({ _id: user.posts });
+    await user.updateOne({ posts: [] });
+    return res.status(200).json({ success: true, data: deletedPosts });
+  } catch (error) {
+    return res.status(500).send({ message: `Server error: ${error.message}` });
+  }
+};
```

Comment:
I think this task also needs an api to delete one of the save in the array by the id input